Library/Homebrew/cleaner.rb

```diff
@@ -48,7 +50,7 @@ def clean
     # [1]: https://github.com/Homebrew/brew/pull/11597
     # [2]: https://github.com/Homebrew/homebrew-core/issues/100190
     # [3]: https://github.com/Homebrew/brew/pull/13215
-    Dir.glob(@formula.info/"**/dir").each do |file|
+    Dir.glob("#{@formula.info}/**/dir").each do |file|
```

Comment:
Why is this change necessary?

Reply:
Add

```ruby
sig { returns(Formula) }
attr_reader :formula
```

instead of accessing the `@formula` variable directly, in case that's the reason.

Reply:
It's because `Dir.glob` needs strings - https://ruby-doc.org/core-3.1.0/Dir.html#method-c-glob

An alternative would be `Pathname.glob` perhaps removing next line and doing,
```suggestion
    @formula.info.glob("**/dir").each do |info_dir_file|
```
Or
```suggestion
    Pathname.glob(@formula.info/"**/dir").each do |info_dir_file|
```

Reply:
You can also do

```ruby
@formula.info.glob("**/dir")
```

Reply:
You can also do

```suggestion
    @formula.info.glob("**/dir").each do |file|
```

Reply:
That was what I was considering but also removing next line. `Pathname#glob` yields `Pathname`s so there is no need to do
```rb
      info_dir_file = Pathname(file)
```